README.md

```diff
@@ -1,38 +1,88 @@
 
-Requirements
+If you use Ansible for Azure resource provisioning purpose, you're strongly encouraged to install this role. 
+
+Prerequisite
+------------
+
+The usage of this role assumes that you've already setup an Ansible environemnt for Azure. For details, please reference Ansible tutorial [Getting Started with Azure](http://docs.ansible.com/ansible/latest/guide_azure.html)
+
+
+Installation
 ------------
 
-Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here. For instance, if the role uses the EC2 module, it may be a good idea to mention in this section that the boto package is required.
+1. Install the role.
+
+  ``` bash
+  $ ansible-galaxy install Azure.azure_preview_modules
+  ```
+
+2. Install Azure Python SDKs.
+
+    Several reasons for installing Python SKDs here.
+
+    - New module is added to the role and this module is for one new Azure service, which is not part of existing Ansible release yet. Corresponding SDK of this new service needs to be installed.
+
+    - Newer version of SDK introduces breaking API change. One specific working version should be installed here.
+
+    The required SDKs are listed in the *[~/files/requirements-azure.txt](files/requirements-azure.txt)* file. The tricky part is the installation location, which has to be the same as where existing Azure Python SKDs installed.
+
+    Taking Ubuntu for example, the existing SDKs can be in folders like
+    `/home/your-user-name/.local/lib/python2.7/site-packages` or `/usr/local/lib/python2.7/dist-packages`. The former is a user folder and the later is a system folder, which requires sudo access. This depends on how you have installed `ansible`. In short, you should install the SDKs the same way as you installed `ansible` so that the SDKs are in the same `site-packages` folder.
+
+    One trick to figure out the correct `site-packages` path is to check the details of existing packages, say `azure-mgmt-storage` by running below command.
+
+      ``` bash
+      $ pip show azure-mgmt-storage
+      ```
+
+    The output can be like below. 
+
+    ``` bash
+    your-user-name@ansible:~$ pip show azure-mgmt-storage
+    Name: azure-mgmt-storage
+    Version: 1.5.0
+    Summary: Microsoft Azure Storage Management Client Library for Python
+    Home-page: https://github.com/Azure/azure-sdk-for-python
+    Author: Microsoft Corporation
+    Author-email: azpysdkhelp@microsoft.com
+    License: MIT License
+    Location: /home/your-user-name/.local/lib/python2.7/site-packages
+    Requires: azure-common, azure-mgmt-nspkg, msrestazure
+    ```
+
+    So, you know that the SDKs are installed in *user* `site-packages` folder. Then, we can use below command to install the listed packages. You can find the `requirements-azure.txt` file in your installed role folder, which is usually at `~\.ansible\roles` folder.
```

Comment:
"So, you know" sounds a kind of casual. maybe something like "From the information above you can learn that".
"... folder. Then, ..." -> "folder and use below command..."
 in general, switching between "you" and "we" should be avoided.
Actually "we" should be avoided.
So instead of "You can find XXX in" it's better  to write "XXX can be found in"